uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -1176,6 +1186,30 @@ define([
       }
 
       /**
+       * Fire and event to start Uploading attachment and resolves it according to the result
+       *
+       * @returns {Promise}
+       */
+      function uploadAttachment () {
+        var deferred = $q.defer();
+
+        var successEvent = $rootScope.$on('uploadFiles: success', function () {
```

Comment:
Logic has changed, now we have `initiateChildProcess` method. I think its better to keep them here as we have other related variables enclosed, which otherwise needs to be in the global scope.